.circleci/base_config.yml

```diff
@@ -261,6 +261,40 @@ jobs:
                   - 3rdParty/iresearch/tests/resources
                   - 3rdParty/rta-makedata
 
+  compile-mac:
+    parameters:
+      preset:
+        type: string
+      build-tests:
+        type: boolean
+      publish-artifacts:
+        type: boolean
+        default: true
+    macos:
+      xcode: 14.2.0
+    resource_class: macos.m1.medium.gen1
+    steps:
+      - checkout-arangodb:
+          with-submodules: true
+          destination: "~/project"
+      - run:
+          name: "Install"
+          command: |
+            brew install cmake openssl@3 llvm@14
+      - run:
+          name: "Configure"
+          command: |
+            export PATH="/opt/homebrew/opt/llvm@14/bin:$PATH"
+            export MACOSX_DEPLOYMENT_TARGET=13.2
+            mkdir ~/build
+            cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -S ~/project -B ~/build
```

Comment:
For other platforms, we can build in various configurations, e.g. Release vs. Debug, Enterprise vs. Community etc.

I guess here we intentionally do not care about `CMAKE_BUILD_TYPE`, `USE_ENTERPRISE`, `USE_STRICT_OPENSSL_VERSION`, `USE_MAINTAINER_MODE` and other CMake options, as we are only checking whether the client tools compile on macOS or not.
Probably adding `USE_ENTERPRISE` is necessary because this flag will have an influence on whether or not we are building arangobackup.

Reply:
Yes because its compiling mac tools, i felt those didn't make sense.
Ill make the USE_ENTERPRISE changes

Reply:
now build using enterprise-pr and community-pr preset for both m1 and x86